code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityMap.java

```diff
@@ -260,7 +250,7 @@ Map<String, Object> asXDMMap(final boolean allowEmpty) {
 			identityMap.put(namespace, namespaceIds);
 		}
 
-		if (!identityMap.isEmpty() || !allowEmpty) {
+		if (!identityMap.isEmpty() || allowEmpty) {
 			xdmMap.put(IdentityConstants.XDMKeys.IDENTITY_MAP, identityMap);
```

Comment:
If the map is not empty, or if `allowEmpty` is true, then add top-level "identityMap" object.  Same as iOS implementation, https://github.com/adobe/aepsdk-edgeidentity-ios/blob/main/Sources/IdentityProperties.swift#L136